scripts/irods/test/test_ifsck.py

```diff
@@ -0,0 +1,45 @@
+import os
+import sys
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+from . import session
+from .. import test
+from .. import lib
+
+class Test_Ifsck(session.make_sessions_mixin([('otherrods', 'rods')], []), unittest.TestCase):
+
+    def setUp(self):
+        super(Test_Ifsck, self).setUp()
+        self.admin = self.admin_sessions[0]
+
+    def tearDown(self):
+        super(Test_Ifsck, self).tearDown()
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_ifsck_has_integer_overflow__issue_4391(self):
+        # Create a file that is exactly 2G in size.
+        file_2g = '2g_issue_4391.txt'
+        lib.make_file(file_2g, 2147483648, 'arbitrary')
+
+        # Create another file that is one byte smaller than 2G in size.
+        file_just_under_2g = 'just_under_2g_issue_4391.txt'
+        lib.make_file(file_just_under_2g, 2147483647, 'arbitrary')
```

Comment:
I am just now noticing that there are several places in our tests where we are calling `lib.make_file` and then failing to unlink/remove the file at the end of the test. So, it would be nice to have a couple of `os.unlink` calls at the end of this test for these two files (especially given the size).  Not so big of a deal for CI/docker but I thought I'd point it out just in case. #1887 may be the place for addressing these types of concerns...

Reply:
Agreed - for completeness... but, note that if we never leave the 'session' collection, then they all get cleaned up with the session 'irm -rf' cleanup.

Reply:
Sorry, I should have clarified that I meant the file on disk that we are `iput`-ing - which is separate from the file created in the vault for the data object represented in iRODS - is not being cleaned up. Another example is `test_rebalance_different_sized_replicas__3486`. I have a local file called "test_rebalance_different_sized_replicas__3486" leftover in the directory from which the test was run. I end up having to clean up the file myself later. Like I said, not a big deal, and actually doesn't affect the result of running the same tests over and over in these two cases.

Reply:
Okay - yes, I thought those all were/are being cleaned up.   We should make the tests leave no trace.